eng/scripts/Language-Settings.ps1

```diff
@@ -8,9 +8,35 @@ $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=con
 
 function Get-dotnet-PackageInfoFromRepo ($pkgPath, $serviceDirectory, $pkgName)
 {
-  $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
-  if (Test-Path $projectPath)
+  $projDirPath = (Join-Path $pkgPath "src")
+
+  if (!(Test-Path $projDirPath))
+  {
+    return $null
+  }
+
+  if ($pkgName)
+  {
+    $projectPath = Join-Path $projDirPath "$pkgName.csproj"
+  }
+  else
+  {
+    $projectPaths = (Resolve-Path (Join-Path $projDirPath "*.csproj")).path
+    if ($projectPaths.Count -gt 1)
+    {
+      LogWarning "There is more than on csproj file in the projectpath/src directory. First project picked."
+      $projectPath = $projectPaths[0]
+    }
+    else {
+      $projectPath = $projectPaths
+    }
+  }
+
+
+
+  if ($projectPath -and (Test-Path $projectPath))
   {
+    $pkgName = Split-Path -Path $projectPath -LeafBase 
```

Comment:
While this is the default package name it is possible for projects to override this in the project itself. I don't think we have any such cases currently but it is something to think about in the future. One of the reasons it might happen for example is if we hit long path issues.

Reply:
Actually you are right. This will be bad for cognitiveservices where we did exactly that.